whois-commons/src/main/java/net/ripe/db/whois/common/grs/RipeAuthoritativeResourceImportTask.java

```diff
@@ -0,0 +1,68 @@
+package net.ripe.db.whois.common.grs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Sets;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import net.ripe.db.whois.common.dao.ResourceDataDao;
+import net.ripe.db.whois.common.scheduler.DailyScheduledTask;
+import org.glassfish.jersey.client.ClientProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import java.io.IOException;
+
+@Component
+public class RipeAuthoritativeResourceImportTask extends AbstractAutoritativeResourceImportTask implements DailyScheduledTask {
+
+    protected static final String TASK_NAME = "RipeAuthoritativeResourceImport";
+
+    private final static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final String baseUrl;
+
+    private Client client;
+
+    @Autowired
+    public RipeAuthoritativeResourceImportTask(final ResourceDataDao resourceDataDao,
+                                               @Value("${grs.import.enabled}") final boolean enabled,
+                                               @Value("${rsng.base.url}") final String baseUrl) {
+        super(enabled, resourceDataDao);
+        this.baseUrl = baseUrl;
+        this.client = ClientBuilder.newBuilder()
+                .property(ClientProperties.CONNECT_TIMEOUT, 10_000)
+                .property(ClientProperties.READ_TIMEOUT, 10_000)
+                .build();
+    }
+
+    /**
+     * Run every 15 minutes.
+     */
+    @Override
+    @Scheduled(cron = "0 5/15 * * * *")
+    @SchedulerLock(name = TASK_NAME)
+    public void run() {
+        doImport(Sets.newHashSet("ripe"));
+    }
+
+    @Override
+    protected AuthoritativeResource fetchAuthoritativeResource(String sourceName) throws IOException {
+        final AuthoritativeResource authoritativeResource = new AuthoritativeResourceJsonLoader(LOGGER).load(
+            OBJECT_MAPPER.readValue(
```

Comment:
Why are you using an explicit ObjectMapper, rather than creating a domain class for the response (and have JAX-RS handle it automatically)?

Reply:
As discussed during standup I completely agree it should be done automatically (I'd set it up like that initially) but for some reason it doesn't work with Jersey. I'll put a TODO in there as agreed.